bin/check_version

```diff
@@ -0,0 +1,77 @@
+#!/bin/sh
+
+if test "x$UPGRADE_MSG" = "x"; then
+    printf >&2 "fatal: Please set '"'$UPGRADE_MSG'"' before sourcing this script\n"
+    exit 1
+fi
+
+die_upgrade() {
+    printf >&2 "fatal: %s\n" "$@"
+    printf >&2 "=> %s\n" "$UPGRADE_MSG"
+    exit 1
+}
+
+major_number() {
+    vers="$1"
+
+    # Hack around 'expr' exiting with code 1 when it outputs 0
```

Comment:
@whyrusleeping and @mateon1 I am not sure what you think is dump.
My opinion is that `expr` exiting with code 1 when it returns 0 is really dump indeed!
It looks like it comes from the fact that it can be used for arithmetics.